### DISCUSSION TITLE
WinRT equivalent of IApplicationActivationManager

### DISCUSSION THREAD

**Discussion by lb90** [microsoft/WindowsAppSDK#2747] [Q&A] [answered]
In order to activate UWP apps we can use the [Application Activation Manager](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-iapplicationactivationmanager) coclass introduced in Windows 8. I wonder if there's a modern WinRT-based replacement API available for launching UWP apps. Thanks!

**Comment by castorix** [comment, marked as answer]
A way is with [AppListEntry.LaunchAsync](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.core.applistentry.launchasync?view=winrt-22621)
You can get AppListEntry from
[PackageManager.FindPackagesForUser](https://docs.microsoft.com/en-us/uwp/api/windows.management.deployment.packagemanager.findpackagesforuser?view=winrt-22621) 
[Package.GetAppListEntriesAsync](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.package.getapplistentriesasync?view=winrt-22621#windows-applicationmodel-package-getapplistentriesasync)

**Reply by lb90** [reply]
Thanks, that's very helpful!

**Reply by DrusTheAxe** [reply]
BTW there's also [.GetAppListEntries()](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.package.getapplistentries) if you prefer a non-async form, though that's only available since 10.0.19041.0 (version 2004 aka 20H1).

**Reply by michael-hawker** [reply]
This was a great helper, just wish this was easier to discover/use, had to update my TFM to get access to the APIs.

I guess there's no easy way to get a reference to the spawned process, eh?

**Reply by DrusTheAxe** [reply]
> > I guess there's no easy way to get a reference to the spawned process, eh?

Only way I know to get the process id of the spawned process is AAM

Or CreateProcess, but that depends
* If the packaged app has an AppExecutionAlias you can CreateProcess("%LOCALAPPDATA%\Microsoft\WindowsApps\alias.exe"...) -- or %LOCALAPPDATA%\Microsoft\WindowsApps\...pkgfamilyname...\alias.exe. CreateProcess' `PROCESS_INFORMATION` should give you the process id and HANDLE
* If the packaged app **is installed with an External Location** and defines `<msix...>` in its Fusion manifest then CreateProcess(....extloc...\foo.exe) should activate the app and again, CreateProcess' `PROCESS_INFORMATION` is at your fingertips

But if you're starting with an ApplicationUserModelID (AUMID) and you need the process id then AAM is the only option I'm aware of.

Why do you ask?
